Add usage to test.sh

geomacy opened the pull request.

geomacy committed:
Add usage to test.sh

lloyddave commented:
It would be beneficial if the script used BRCLI_HOME env var to avoid over-writing existing credentials.  Maybe use /tmp or /tmp/<user> as the directory?

lloyddave commented on test/test.sh:
```diff
@@ -141,25 +141,40 @@ function shouldStartEntity() {
+    local brooklyn_url=${1}
     local user=${2}
     local password=${3}
 
```
If shouldHaveBr fails the script should exit.

lloyddave commented on test/test.sh:
```diff
@@ -141,25 +141,40 @@ function shouldStartEntity() {
 function main() {
 
     local application_yaml=test_app.yaml
     local appname=$(grep name: ${application_yaml} | sed 's/.*: *//')
```
A YAML file may have multiple lines that match 'name: '.  Suggest that you match on the first one found with:
`grep name: ${application_yaml} | head -1 | sed 's/.*: *//'`
Or possibly retrieve the name from the output of the deploy command?

lloyddave replied:
The statement for reading appname should be moved to after processing args in case a different YAML file is specified.

lloyddave replied:
The tests are very dependent on the sample blueprint.  Would it be better to include the blueprint in the test script and deploy it to localhost?  
Or maybe it could be made more generic by doing later operations on the first or last entity of an application rather than assuming TomcatServer?

geomacy committed:
Updates to test.sh from review.

For simplicity have removed the ability to pass in a test file.
Tests are dependent on using Tomcat, so simplify things by having the
code explicitly make use of this knowledge where appropriate.

Added use of BRCLI_HOME variable.

geomacy committed:
Fix usage line.

lloyddave commented:
Script looks fine, but the YAML file needs to be changed to have a location of something like 'SetToValidLocation'.

geomacy committed:
Fix name of test location to indicate it is a placeholder.

geomacy commented:
I have updated the test yaml to make the location an obvious placeholder.

lloyddave commented:
Tests pass against byon hosts and softlayer (after updating the max timeout).  Merging.
